src/class.c

```diff
@@ -1613,6 +1615,31 @@ void InterfaceDeclaration::finalizeSize(Scope *sc)
 {
     structsize = Target::ptrsize * 2;
     sizeok = SIZEOKdone;
+
+    // set the offset of base interfaces from this (most derived) interface.
+    unsigned offset = 0;
+    //if (vtblInterfaces->dim) printf("\n%s->finalizeSize()\n", toChars());
+    for (size_t i = 0; i < vtblInterfaces->dim; i++)
+    {
+        BaseClass *b = (*vtblInterfaces)[i];
+        unsigned thissize = Target::ptrsize;
+
+        alignmember(STRUCTALIGN_DEFAULT, thissize, &offset);
+        b->offset = offset;
+        //printf("\tvtblInterfaces[%d] b->sym = %s, offset = %d\n", i, b->sym->toChars(), b->offset);
+
+        // Take care of single inheritance offsets
+        while (b->baseInterfaces_dim)
+        {
+            b = &b->baseInterfaces[0];
+            b->offset = offset;
+            //printf("\tvtblInterfaces[%d] +  sym = %s, offset = %d\n", i, b->sym->toChars(), b->offset);
+        }
+
+        offset += thissize;
+        if (alignsize < thissize)
+            alignsize = thissize;
+    }
```

Comment:
This seems very much a copy of the same code in ClassDeclaration. Move to a function?